central/declarativeconfig/health/datastore/datastore_impl.go

```diff
@@ -66,3 +78,11 @@ func (ds *datastoreImpl) UpdateErrorMessageForDeclarativeConfig(ctx context.Cont
 
 	return ds.UpsertDeclarativeConfig(ctx, existingHealth)
 }
+
+func (ds *datastoreImpl) verifyContextKey(ctx context.Context) error {
+	if !declarativeconfig.HasModifyDeclarativeResourceKey(ctx) {
+		return errox.NotAuthorized.New(
+			"declarative config health cannot be modified with the current permission")
```

Comment:
maybe add some hint what permissions are required and what are current. This could be solve in docs